### DISCUSSION TITLE
Does this Library work with MongoDb databases?

### DISCUSSION THREAD

**Discussion by KDCheapCheap** [JonPSmith/AuthPermissions.AspNetCore#85] [Q&A] [answered]
May be a bit of a noob question but I can see in the SetupAspNetCoreAndDatabase section of the Wiki, you call UsingEfCoreSqlServer(connectionString). Is this just for example purposes or does it only work with SQL?

**Comment by JonPSmith** [comment, marked as answer]
Sorry, but the AuthP library doesn't support the MongoDb database. That's because it uses EF Core to access the database and EF Core doesn't support the MongoDb database.

Currently AuthP supports SqlServer and Postgres, but version 5 of the AuthP library (which will be out in a month or so, will allow you to many more EF Core databases providers.

**Reply by KDCheapCheap** [reply]
Ahh okay, thanks for the info and quick reply! Hopefully v5 brings Mongo into the fold too!

**Reply by JonPSmith** [reply]
I can't add Mongo to the AuthP library because its not supported by EF Core. But you might like to read some of of the articles about how to build multi-tenant apps, starting with [this article](https://www.thereformedprogrammer.net/building-asp-net-core-and-ef-core-multi-tenant-apps-part1-the-database/)
